homeassistant/components/modbus/validators.py

```diff
@@ -127,25 +126,27 @@ def scan_interval_validator(config: dict) -> dict:
 
             for entry in hub[conf_key]:
                 scan_interval = entry.get(CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL)
-                if scan_interval < MINIMUM_SCAN_INTERVAL:
-                    if scan_interval == 0:
-                        continue
+                if scan_interval == 0:
+                    continue
+                if scan_interval < 5:  # there are not constant in core for this.
                     _LOGGER.warning(
-                        "%s %s scan_interval(%d) is adjusted to minimum(%d)",
+                        "%s %s scan_interval(%d) is lower than HA standard",
```

Comment:
I like the fact how you left the warning in, as in the current situation, it may lead to stacking quite a bit of tasks (which at that point, is probably not just modbus).

A small suggestion to improve the test a little:

```suggestion
                        "%s %s scan_interval(%d) is lower than 5 seconds, "
                        "which may cause Home Assistant stability issues",
```

Reply:
A lot better wording.